### ISSUE TITLE
Open cert Renderer and OpenCerts 2.0 DNS-TXT Verification

### ISSUE THREAD

**Issue by mohammedkalimullah** [OpenCerts/opencerts-documentation#85]
Hi Support,

we are hosting the renderer site in our own DC.

we would like to check on the below items.

1.Can we allow our renderersite to access only by opencert? if yes can we have opencert ip range. so that we can whitelist.
2. Regarding DNS-Text verification, Please advise is this step is Mandatory. if yes do we need to change our certificate templates? is this field is in opencert schema?

**Comment by Nebulis** [comment]
Hi,

1. You can use https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options to prevent rendering in non-authorized frame
2. Is your question `Do we need to perform DNS check on the renderer ?`. The answer is no

**Comment by Nebulis** [comment]
Hi,

1. You can use https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options to prevent rendering in non-authorized frame
2. I dont understand your question, could you rephrase ?

**Comment by mohammedkalimullah** [comment]
Hi,

we have hosted our renderer site in our DC. 

we would like to secure the renderer site to be able to access only by Opencert. This can be done whitelisting opencert ip. Can we have opencert ip range? so that we can open from our firewall.

2. Is this step OpenCerts 2.0 DNS-TXT Verification is Mandatory or optional?

Please advise if we need to change our certificate templates? is this field is in opencert schema?

**Comment by Nebulis** [comment]
Please read the link I shared to whitelist from frames host.

I'm still not sure about your second question. Do you ask if you need to check for Dns Redbird to make sure the certificate is valid ? If that's your question then your don't need to

**Comment by mohammedkalimullah** [comment]
Hi ,

our renderer site is hosted on-premise e.g https://demo-renderer.opencerts.io

we would like to prevent the renderer site from unauthorized access/attack. so we would like to whitelist opencert ip and allow the site accessible only from opencert

to achieve this we need open cert ip range. Please assist.

For the second question, yes for issuing and verifying the certificate do we need to implement this step OpenCerts 2.0 DNS-TXT Verification?
https://docs.opencerts.io/dns_verification.html

**Comment by Nebulis** [comment]
The verification is happening on opencerts. 
If your certificates follow the 2.0 schema and you need DNS verification, you need to add it into your certificates.
No verification is needed from decentralized renderer.

**Comment by mohammedkalimullah** [comment]
Thanks!! 
How about the question 1? can we have opencert ip to whitelist?

**Comment by Nebulis** [comment]
Request are made from the client (browser) as such there is nothing to whitelist

**Comment by mohammedkalimullah** [comment]
Hi,
I am not sure have you got my question correct.

we do not want to expose our renderer site to public and allow only for OPencerts. This is an security concern. 
if we allow, there could be potential security risk of getting ddos and other cyber attack.
so we are requesting to provide Opencert ip. so that we can whitelist only Opencert ip to allow access our site.

Hope you understand.

**Comment by Nebulis** [comment]
I understand correctly:
- Requests are made from the client (browser) as such there is nothing to whitelist
- If you want to make sure your renderer is only use in opencerts:
  1. disallow usage out of frame
  2. use developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

**Comment by mohammedkalimullah** [comment]
Yes. request is made from web browser. The x-frame is to prevent clickjacking only. Meaning the iframe will not be loaded however this will not prevent other cyber attacks.
I can set this in my site. X-Frame-Options: allow-from https://opencert.io/ 
But we would like to prevent attack from network layer itself.
